app/jobs/search_deindexer_job.rb

```diff
@@ -0,0 +1,10 @@
+# encoding: utf-8
+
+class SearchDeindexerJob < ApplicationJob
+  queue_as :cms_search_indexer
+
+  def perform(class_name, id)
```

Comment:
This may depend on the model, and if it's a real or soft-delete.  But generally ... by the time this job gets run, the ActiveRecord model isn't accessible.  (Also that global-id stuff does a `ModelClass.find(id)`, which isn't going to include `with_deleted` anyways).

Reply:
Metaprogramming++